FrontEndReact/src/View/Login/SetNewPassword.js

```diff
@@ -3,87 +3,277 @@ import 'bootstrap/dist/css/bootstrap.css';
 import ErrorMessage from '../Error/ErrorMessage.js';
 import { genericResourcePUT, validPasword } from '../../utility.js';
 import Login from './Login.js';
+import { Button, TextField, FormControl, Box, Typography } from '@mui/material';
+import ErrorOutlineIcon from '@mui/icons-material/ErrorOutline';
+import CheckIcon from '@mui/icons-material/Check';
+
+
 
 class SetNewPassword extends Component {
     constructor(props) {
         super(props);
+
         this.state = {
             errorMessage: null,
-            passSet: false
+            passSet: false,
+            password: '',
+            password2: '',
+
+            errors : {
+                password: '',
+                password2: ''
+            },
+
+            PasswordStrength : {
+                STRONG: 'STRONG',
+                MEDIUM: 'MEDIUM',
+                WEAK: 'WEAK'
+            }
+        }
+
+        this.handleChange = (e) => {
+            const { id, value } = e.target;
+
+            this.testingPasswordStrength(value)
+
+            this.setState({
+                [id]: value,
+                errors: {
+                    ...this.state.errors,
+                    [id]: value.trim() === '' ? `${id.charAt(0).toUpperCase() + id.slice(1)} cannot be empty` : '',
+                },
+            });
+        };
+
+        this.getIcon = (strength) => {
+            let icon = ErrorOutlineIcon;
+        
+            switch(strength) {
+                default: 
+                    icon = ErrorOutlineIcon;
+                    break;
+                case 'WEAK':
+                    icon = ErrorOutlineIcon;
+                    break;
+                case 'MEDIUM':
+                    icon = ErrorOutlineIcon;
+                    break;
+                case 'STRONG':
+                    icon = CheckIcon;
+                    break;
+            }
+
+            return icon;
+        };
+
+        this.generateColors = (strength) => {
+            let result = [];
+
+            const COLORS = {
+                NEUTRAL: '#E2E2E2',
+                WEAK: '#B40314',
+                MEDIUM: '#D39323',
+                STRONG: '#7B927F',
+            };
+        
+            switch(strength) {
+                default: 
+                    result = [COLORS.WEAK, COLORS.NEUTRAL, COLORS.NEUTRAL, COLORS.NEUTRAL];
+                    break;
+                case 'WEAK':
+                    result = [COLORS.WEAK, COLORS.NEUTRAL, COLORS.NEUTRAL, COLORS.NEUTRAL];
+                    break;
+                case 'MEDIUM':
+                    result = [COLORS.MEDIUM, COLORS.MEDIUM, COLORS.NEUTRAL, COLORS.NEUTRAL];
+                    break;
+                case 'STRONG':
+                    result = [COLORS.STRONG, COLORS.STRONG, COLORS.STRONG, COLORS.STRONG];
+                    break;
+            }
+        
+            return result;
+        };
+
+        this.testingPasswordStrength = (password) => {
```

Comment:
maybe typo: did you mean `testPasswordStrength`?

Reply:
I committed changes to fix the typo!